bundles/mapping/mapwfs2/service/Mediator.js

```diff
@@ -395,7 +395,11 @@ Oskari.clazz.category('Oskari.mapframework.bundle.mapwfs2.service.Mediator', 'ge
         } else {
             // FIXME: pass coordinates from server in response, but not like this
             data.data.lonlat = this.lonlat;
-            me.WFSLayerService.emptyWFSFeatureSelections(layer);
+
+            if (me.WFSLayerService.getWFSSelections().length > 0) {
+                me.WFSLayerService.emptyWFSFeatureSelections(layer);
```

Comment:
This if should be inside the emptyWFSFeatureSelections() implementation. Otherwise looks good.